docs/kubernetes.aad.md

```diff
@@ -0,0 +1,100 @@
+
+## Prerequision
+1. An Azure Active Directory tenant, will refer as `AAD Tenant`. You can use the tenant for your Azure subscription;
+2. A `Web app / API` type AAD application, will refer as `Server Application`. This application represents the `apiserver`;
+3. A `Native` type AAD application, will refer as `Client Application`. This application is for user login via `kubectl`. You'll need to add delegated permission to `Server Application`, please see [troubleshooting](#loginpageerror) section for detail.
+
+## Deployment
+Follow the [deployment steps](kubernetes.md#deployment). In step #4, add the following under 'properties' section:
+```
+"aadProfile": {
+    "serverAppID": "",
+    "clientAppID": "",
+    "tenantID": ""
+}
+```
+
+- `serverAppID`   : the `Server Application`'s ID
+- `clientAppID`   : the `Client Application`'s ID
+- `tenantID`      : (optional) the `AAD tenant`'s ID. If not specified, will use the tenant of the deployment subscription.
+
+After template generation, the local generated kubeconfig file (`_output/<instance>/kubeconfig/kubeconfig.<location>.json`) will have the default user using AAD.
+Initially it isn't assoicated with any AAD user yet. To get started, try any kubectl command (like `kubectl get pods`), and you'll be prompted to the device login process. After login, you will be able to operate the cluster using your AAD identity.
+
+### Note
+Please note that as of Kubernetes 1.7, the default is authorization mode is `AlwaysAllow`, which means any authenticated user have full access of the cluster.
+OpenID Connect is an authentication protocol responsible for identify users only, so initally all active accounts under the tenant will be able to login and have full admin privilege of the cluster.
+
+In this case you may want to also turn on RBAC for your cluster. Following instructions are for turnning on RBAC together with AAD integration:
+
+1. Since we use AAD object ID as OpenID Connect identity.
+    You'll first need to figure out your account's object ID. Here is how to do it using Azure Portal:
+    Navigate to `Azure Active Directory` -> `Users and groups` -> `All users`. And choose your account in right pannel. Switch to `Manage` -> `Profile`, and you can see the `Object ID` property.
+2. Figure out your user name. The user name would be in form of `IssuerUrl#ObjectID` format.
+    You can navigate to `https://login.microsoftonline.com/{tenantid}/.well-known/openid-configuration`, and find the `IssuerUrl` under `issuer` property.
+3. Add your account as admin role
+```
+kubectl create clusterrolebinding aad-default-cluster-admin-binding --clusterrole=cluster-admin --user={UserName}
+```
+4. Turn on RBAC on master nodes.
```

Comment:
I've put a link to the rbac doc. And add a note for that this section is for how to enable rbac manually.